src/Streamly/Internal/LZ4.hs

```diff
@@ -242,6 +242,8 @@ compressChunk i ctx arr = do
               let dstEnd = dstBegin `plusPtr`
                             ((fromIntegral :: CInt -> Int) compLen + 8)
                   compArr = MArray.Array fptr dstEnd dstMax
+              -- It is safe to shrink here as we need to hold the last 64KB of
+              -- the previous uncompressed array and not the compressed one.
```

Comment:
This comment is not needed, as there is no caution required, you may or may not shrink it does not matter.